src/app/vm/vm-sidebar/affinity-group-selector/affinity-group-selector-container.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { AffinityGroup, AffinityGroupType } from '../../../shared/models';
+import { select, Store } from '@ngrx/store';
+import { State } from '../../../reducers';
+import * as affinityGroupActions from '../../../reducers/affinity-groups/redux/affinity-groups.actions';
+import * as affinityGroupSelectors from '../../../reducers/affinity-groups/redux/affinity-groups.selectors';
+
+@Component({
+  selector: 'cs-affinity-group-container-selector',
+  template: `
+    <cs-affinity-group-selector
+      [affinityGroups]="affinityGroups$ | async"
+      [sortedAffinityGroups]="sortedAffinityGroups$ | async"
+      [enablePreselected]="enablePreselected"
+      [preselectedAffinityGroups]="preselectedAffinityGroups"
+      (createdAffinityGroup)="createAffinityGroup($event)"
+      (canceled)="cancel()"
+      (submited)="submit($event)"
+    ></cs-affinity-group-selector>
+  `,
+})
+export class AffinityGroupSelectorContainerComponent {
+  readonly affinityGroups$;
+  readonly sortedAffinityGroups$;
+  public types = [AffinityGroupType.antiAffinity, AffinityGroupType.affinity];
+  public preselectedAffinityGroups: AffinityGroup[];
+  public enablePreselected: boolean;
+
+  constructor(
+    private store: Store<State>,
+    private dialogRef: MatDialogRef<AffinityGroupSelectorContainerComponent>,
+    @Inject(MAT_DIALOG_DATA) public data,
+  ) {
+    this.enablePreselected = data.enablePreselected;
+    this.preselectedAffinityGroups = data.preselectedAffinityGroups || [];
+
+    this.sortedAffinityGroups$ = this.store.pipe(
```

Comment:
We decided to initialize variables inline instead of in constructor. https://docs.google.com/document/d/1w_E7JjI60kLOoygIOh-1kzNaMeots_o_IoUCLI4Paxo/edit#

Reply:
I did it, cause I use in this selectors preselectedAffinityGroup, which I define in a constructor